dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,149 +91,540 @@ def _index_in_schema(index, schema):
-            ]
-            if partitions and dataset.metadata:
-                # Dont use dataset.metadata for partitioned datasets, unless
-                # the column-chunk metadata includes the `"file_path"`.
-                # The order of dataset.metadata.row_group items is often
-                # different than the order of `dataset.pieces`.
-                if not col_chunk_paths or (
-                    len(dataset.pieces) != dataset.metadata.num_row_groups
+
+def _generate_dd_meta(schema, index, categories, partition_info):
+    partition_obj = partition_info["partitions"]
+    partitions = partition_info["partition_names"]
+    columns = None
+
+    has_pandas_metadata = schema.metadata is not None and b"pandas" in schema.metadata
+
+    if has_pandas_metadata:
+        pandas_metadata = json.loads(schema.metadata[b"pandas"].decode("utf8"))
+        (
+            index_names,
+            column_names,
+            storage_name_mapping,
+            column_index_names,
+        ) = _parse_pandas_metadata(pandas_metadata)
+        if categories is None:
+            categories = []
+            for col in pandas_metadata["columns"]:
+                if (col["pandas_type"] == "categorical") and (
+                    col["name"] not in categories
                 ):
-                    dataset.schema = dataset.metadata.schema
-                    dataset.metadata = None
-        else:
-            partitions = []
-
-        # Statistics are currently collected at the row-group level only.
-        # Therefore, we cannot perform filtering with split_row_groups=False.
-        # For "partitioned" datasets, each file (usually) corresponds to a
-        # row-group anyway.
-        # TODO: Map row-group statistics onto file pieces for filtering.
-        #       This shouldn't be difficult if `col_chunk_paths==True`
-        if not split_row_groups and not col_chunk_paths:
-            if gather_statistics is None and not partitions:
-                gather_statistics = False
-                if filters:
-                    raise ValueError(
-                        "Filters not supported with split_row_groups=False "
-                        "(unless proper _metadata is available)."
-                    )
-            if gather_statistics and not partitions:
-                raise ValueError(
-                    "Statistics not supported with split_row_groups=False."
-                    "(unless proper _metadata is available)."
-                )
+                    categories.append(col["name"])
+    else:
+        index_names = []
+        column_names = schema.names
+        storage_name_mapping = {k: k for k in column_names}
+        column_index_names = [None]
+
+    if index is None and index_names:
+        index = index_names
+
+    if set(column_names).intersection(partitions):
+        raise ValueError(
+            "partition(s) should not exist in columns.\n"
```

Comment:
This is an odd case! Is there a test for it?

Reply:
No, this doesn't seem to be tested.  The lines actually outdate my involvement with the code, and I'd be happy to yank then if you don't think the check is necessary :)

Reply:
I suppose leave it. I wonder if it's technically disallowed or not: partition splitting is not actually defined in the spec, and the file data in the file might just reflect the same as in the directory-implied value.